pkg/rule/rule.go

```diff
@@ -33,11 +37,114 @@ func (g Group) OriginalFile() string {
 	return g.originalFile
 }
 
+// GetRules returns a list of rules, which is used in UI rules page.
+func (g *Group) GetRules() []rules.Rule {
+	var rs []rules.Rule
+	for _, rule := range g.Rules() {
+		if r, ok := rule.(*rules.AlertingRule); ok {
+			rs = append(rs, &AlertingRule{r, g.PartialResponseStrategy})
+		} else if r, ok := rule.(*rules.RecordingRule); ok {
+			rs = append(rs, &RecordingRule{r, g.PartialResponseStrategy})
+		}
+	}
+	return rs
+}
+
+// GetAlertingRules returns a list of alerting rules, which is used in UI alerts page.
+func (g Group) GetAlertingRules() []*AlertingRule {
+	var alerts []*AlertingRule
+	for _, rule := range g.Rules() {
+		if alertingRule, ok := rule.(*rules.AlertingRule); ok {
+			alerts = append(alerts, &AlertingRule{alertingRule, g.PartialResponseStrategy})
+		}
+	}
+	sort.Slice(alerts, func(i, j int) bool {
+		return alerts[i].State() > alerts[j].State() ||
+			(alerts[i].State() == alerts[j].State() &&
+				alerts[i].Name() < alerts[j].Name())
+	})
+	return alerts
+}
+
 type AlertingRule struct {
 	*rules.AlertingRule
 	PartialResponseStrategy storepb.PartialResponseStrategy
 }
 
+// HTMLSnippet returns a human-readable string representation of alerting rules,
+// decorated with HTML elements for use the web frontend.
+func (r *AlertingRule) HTMLSnippet(pathPrefix string) html_template.HTML {
+	alertMetric := model.Metric{
+		model.MetricNameLabel: "ALERTS",
+		"alertname":           model.LabelValue(r.Name()),
+	}
+
+	labelsMap := make(map[string]string, len(r.Labels()))
+	for _, l := range r.Labels() {
+		labelsMap[l.Name] = html_template.HTMLEscapeString(l.Value)
+	}
+
+	annotations := r.Annotations()
+	annotationsMap := make(map[string]string, len(annotations))
+	for _, l := range annotations {
+		annotationsMap[l.Name] = html_template.HTMLEscapeString(l.Value)
+	}
+
+	ar := ruleFormat{
+		Alert:                   fmt.Sprintf("<a href=%q>%s</a>", pathPrefix+strutil.TableLinkForExpression(alertMetric.String()), r.Name()),
+		Expr:                    fmt.Sprintf("<a href=%q>%s</a>", pathPrefix+strutil.TableLinkForExpression(r.Query().String()), html_template.HTMLEscapeString(r.Query().String())),
+		For:                     model.Duration(r.HoldDuration()),
+		Labels:                  labelsMap,
+		Annotations:             annotationsMap,
+		PartialResponseStrategy: storepb.PartialResponseStrategy_name[int32(r.PartialResponseStrategy)],
+	}
+
+	byt, err := yaml.Marshal(ar)
+	if err != nil {
+		return html_template.HTML(fmt.Sprintf("error marshaling alerting rule: %q", html_template.HTMLEscapeString(err.Error())))
+	}
+	return html_template.HTML(byt)
+}
+
+type RecordingRule struct {
```

Comment:
Let's add a comment or make this unexported

Reply:
Done. Please take a look agian